### PR TITLE
ethdb/pebble: Fix `MemTableStopWritesThreshold`

### PR DESCRIPTION
**Pull request by patrick-ogrady**
`MemTableStopWritesThreshold` was set to the max size of all memtables before blocking writing but should be set to the max number of memtables. This is documented [here](https://github.com/cockroachdb/pebble/blob/master/options.go#L738-L742) and used in the code [here](https://github.com/cockroachdb/pebble/blob/master/db.go#L1892-L1903).

Nice work adding Pebble support! ❤️

### CODE REVIEW

**Comment by patrick-ogrady** [comment]
It may also make sense to specify `WALBytesPerSync` given you set `pebble.NoSync` in `Set`. By default, it is set to 0 (all `NoSync` writes go directly to disk): https://github.com/cockroachdb/pebble/blob/master/options.go#L798-L807.

This may affect crash recovery but can improve performance.

File: ethdb/pebble/pebble.go [hunk 1]
```diff
@@ -157,7 +163,10 @@ func New(file string, cache int, handles int, namespace string, readonly bool) (
...
 		// MemTableStopWritesThreshold places a hard limit on the size
 		// of the existent MemTables(including the frozen one).
-		MemTableStopWritesThreshold: memTableLimit * memTableSize,
+		// Note, this must be the number of tables not the size of all memtables
+		// according to https://github.com/cockroachdb/pebble/blob/master/options.go#L738-L742
+		// and to https://github.com/cockroachdb/pebble/blob/master/db.go#L1892-L1903.
+		MemTableStopWritesThreshold: memTableLimit,
 
 		// The default compaction concurrency(1 thread),
 		// Here use all available CPUs for faster compaction.
```
Comment (patrick-ogrady) [inline ethdb/pebble/pebble.go:172]: You could also change this to just be:
```golang
MaxConcurrentCompactions: runtime.NumCPU,
```

But didn't want to bloat this bugfix PR

Comment (rjl493456442) [reply ethdb/pebble/pebble.go:172]: Oops, thanks for catching it!

File: ethdb/pebble/pebble.go [hunk 2]
```diff
@@ -125,7 +125,13 @@ func New(file string, cache int, handles int, namespace string, readonly bool) (
...
 	}
 	logger := log.New("database", file)
-	logger.Info("Allocated cache and file handles", "cache", common.StorageSize(cache*1024*1024), "handles", handles)
+	logger.Info(
+		"Allocated cache and file handles",
+		"cache",
+		common.StorageSize(cache*1024*1024),
+		"handles",
+		handles,
+	)
```
Comment (rjl493456442) [inline ethdb/pebble/pebble.go:134]: Can you revert this change? I think the original format is better.

Comment (patrick-ogrady) [reply ethdb/pebble/pebble.go:134]: Yep. Sorry, my linter has a mind of its own.

**Review by rjl493456442 — APPROVED** [review]

**Comment by holiman** [comment]
Thanks!
